src/status_im2/contexts/chat/photo_selector/view.cljs

```diff
@@ -0,0 +1,108 @@
+(ns status-im2.contexts.chat.photo-selector.view
+  (:require [i18n.i18n :as i18n]
+            [quo.components.safe-area :as safe-area]
+            [quo2.components.notifications.info-count :as info-count]
+            [quo2.core :as quo2]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [react-native.linear-gradient :as linear-gradient]
+            [reagent.core :as reagent]
+            [status-im2.contexts.chat.photo-selector.style :as style]
+            [status-im.utils.core :as utils]
+            [utils.re-frame :as rf]))
+
+(def selected (reagent/atom []))
+
+(defn bottom-gradient
+  [chat-id selected-images]
+  [:f>
+   (fn []
+     (let [safe-area (safe-area/use-safe-area)]
+       (when (or (pos? (count @selected)) selected-images)
+         [linear-gradient/linear-gradient
+          {:colors [:black :transparent]
+           :start  {:x 0 :y 1}
+           :end    {:x 0 :y 0}
+           :style  (style/gradient-container safe-area)}
+          [quo2/button
+           {:style               {:align-self        :stretch
+                                  :margin-horizontal 20}
+            :on-press            #(do
+                                    (rf/dispatch [:chat.ui/clear-sending-images chat-id])
+                                    (doseq [item @selected]
+                                      (rf/dispatch [:chat.ui/camera-roll-pick item]))
+                                    (reset! selected [])
+                                    (rf/dispatch [:bottom-sheet/hide]))
+            :accessibility-label :confirm-selection}
+           (i18n/label :t/confirm-selection)]])))])
+
+(defn clear-button
+  []
+  (when (pos? (count @selected))
+    [rn/touchable-opacity
+     {:on-press            #(reset! selected [])
+      :style               (style/clear-container)
+      :accessibility-label :clear}
+     [quo2/text {:weight :medium} (i18n/label :t/clear)]]))
+
+(defn image
+  [item index _ {:keys [window-width]}]
+  [rn/touchable-opacity
+   {:active-opacity      1
+    :on-press            (fn []
+                           (if (some #{item} @selected)
+                             (reset! selected (vec (remove #(= % item) @selected)))
+                             (swap! selected conj item)))
+    :accessibility-label (str "image-" index)}
+   [rn/image
+    {:source {:uri item}
+     :style  (style/image window-width index)}]
+   (when (some #{item} @selected)
+     [rn/view {:style (style/overlay window-width)}])
+   (when (some #{item} @selected)
+     [info-count/info-count
+      {:style               style/image-count
+       :accessibility-label (str "count-" index)}
+      (inc (utils/first-index #(= item %) @selected))])])
+
+(defn photo-selector
+  [chat-id]
+  (rf/dispatch [:chat.ui/camera-roll-get-photos 20])
```

Comment:
Is this really correct? This `dispatch` is definitely going to be called multiple times, once for every render. The component shouldn't have side-effect calls in free form like this, they should at least exist inside a useEffect if the re-frame machinery isn't sufficient.

Reply:
This is a functional component. The part before ` [:f> (fn []` gets called once.

Reply:
You're correct about `:f>` :) You could also decouple the decision about when to fetch data from this component by moving the dispatch to:

https://github.com/status-im/status-mobile/blob/8c1af01bf8cc15d839d7846e043e19fa1dc6e4cb/src/status_im/ui2/screens/chat/composer/view.cljs#L301 

But it's fine by me.